src/api/volume.go

```diff
@@ -119,23 +123,16 @@ func (s *VolumeService) Detach(ctx context.Context, volume *csi.Volume, server *
 		return volumes.ErrVolumeNotFound
 	}
 
-	hcloudServer, _, err := s.client.Server.GetByID(ctx, int(server.ID))
-	if err != nil {
-		return err
-	}
-	if hcloudServer == nil {
-		return volumes.ErrServerNotFound
-	}
-
 	if hcloudVolume.Server == nil {
 		return volumes.ErrNotAttached
 	}
-	if hcloudVolume.Server.ID != hcloudServer.ID {
-		return volumes.ErrAlreadyAttached
-	}
 
 	action, _, err := s.client.Volume.Detach(ctx, hcloudVolume)
 	if err != nil {
+		//TODO: update once hcloud upgraded to v1.12
```

Comment:
I think this part can be included in a PR about fixing the locked server errors CSI-driver receives from the hcloud.